src/config_waiter.rs

```diff
@@ -30,14 +30,18 @@ const BOOT_MAX_IDLE: Duration = Duration::from_secs(10);
 /// produced at all times.
 ///
 /// The returned Receiver can be awaited
-pub fn wait_for_partition_boot(config_store: Arc<RwConfigStore>, partition: u16) -> Receiver<()> {
+pub fn wait_for_partition_boot(
+    config_store: Arc<RwConfigStore>,
+    partition: u16,
+    shutdown: CancellationToken,
+) -> Receiver<bool> {
```

Comment:
Not sure if we should use something more explicit than a bool?

Reply:
Enum?

Reply:
Thinking about it and I'm not sure if it's much better? Just like returning a bool from a function really

Reply:
Yeah I'm indifferent.

I like enums because they're a bit more descriptive

like you could have

```
enum BootResult {
    Started,
    Canceled,
}
```

Then this returns a `Reciver<BootResult>` much more clear what it is